nodl_python/nodl/index/_index.py

```diff
@@ -0,0 +1,88 @@
+# Copyright 2020 Canonical Ltd.
+#
+# This program is free software: you can redistribute it and/or modify it under the terms of the
+# GNU Limited General Public License version 3, as published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+# without even the implied warranties of MERCHANTABILITY, SATISFACTORY QUALITY, or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Limited General Public License for more details.
+#
+# You should have received a copy of the GNU Limited General Public License along with
+# this program. If not, see <http://www.gnu.org/licenses/>.
+
+
+from pathlib import Path
+from typing import List, Sequence, TYPE_CHECKING
+
+from ament_index_python.packages import get_package_share_directory
+
+from nodl._parsing._parsing import _parse_multiple
+from nodl.errors import ExecutableNotFoundError, NoNoDLFilesError
+
+if TYPE_CHECKING:  # pragma: no cover
+    from .types import Node  # noqa: F401
+
+
+def _get_nodl_xml_files_in_path(*, path: Path) -> List[Path]:
+    """Return all files with NoDL extension (.nodl.xml) in a path."""
+    return sorted(path.rglob('*.nodl.xml'))
```

Comment:
Ditto here: is there a good reason this needs to be its own function? We're actually adding lines of code, and I'm not sure what we're getting in return. In general, a function should be created either because the logic is reused in multiple places or because it makes the whole thing conceptually easier to understand. I'm not sure either of those are happening here.

Reply:
This is actually used for file completion on the command line

Reply:
Ha! Well then, you can safely ignore me.

Reply:
It was made a function in case we need to change what qualifies as a nodl file (e.g. imposing a requirement on the name beyond .nodl.xml)

Reply:
Makes sense.